onnx/defs/math/defs.cc

```diff
@@ -385,6 +385,37 @@ will throw errors.
   .Output(0, "output", "The softmax normalized output values with the same "
           "shape as input tensor.");
 
+OPERATOR_SCHEMA(LogSoftmax)
+  .NumInputs(1)
+  .NumOutputs(1)
+  .SetDoc(R"DOC(
+The operator computes the logsoftmax normalized values for each layer in the batch
+ of the given input. The input is a 2-D tensor (Tensor<float>) of size
+(batch_size x input_feature_dimensions). The output tensor has the same shape
+and contains the logsoftmax normalized values of the corresponding input.
+
+X does not need to explicitly be a 2D vector; rather, it will be
```

Comment:
I wish we had a way of not copy pasting things like this. It will make the spec harder to edit if we need to change how these mechanics work.